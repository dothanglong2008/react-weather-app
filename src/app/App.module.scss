* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html {
  font-size: calc(10px + 1vmin);
}

ul {
  list-style: none;
}

img,
picture,
figure {
  width: 100%;
  object-fit: none;
}

button,
a {
  cursor: pointer;
}

.app {
  background-color: #ebebeb;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}
