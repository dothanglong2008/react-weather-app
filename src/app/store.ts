import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import weatherReducer from "../features/weather/weatherSlice";
import searchFormReducer from "../components/weather/search/searchFormSlice";

export const store = configureStore({
  reducer: {
    weather: weatherReducer,
    searchForm: searchFormReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
