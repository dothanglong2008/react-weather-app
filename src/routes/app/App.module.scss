$background-color-primary: #ffffff;
$text-color-primary: #000000;
$font-size-primary: 16px;

* {
  margin: 0;
  padding: 0;
  width: 100%;
}

body {
  background-color: $background-color-primary;
  color: $text-color-primary;
  font-size: $font-size-primary;
}

ul {
  list-style: none;
}

img,
picture,
video {
  max-width: 100%;
}
