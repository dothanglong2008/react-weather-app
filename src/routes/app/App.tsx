import Item from "../../components/weather/single-item/item/Item";
import styles from "./App.module.scss";

function App() {
  return (
    <div className={styles.app}>
      <Item />
    </div>
  );
}

export default App;
