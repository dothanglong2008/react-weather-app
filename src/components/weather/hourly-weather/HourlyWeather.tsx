import { useAppSelector } from "../../../app/hooks";
import { selectWeatherData } from "../../../features/weather/weatherSlice";
import useFetchWeatherByLocation from "../../../hooks/useFetchWeatherByLocation";
import SectionHeading from "../../ui/elements/section-heading/SectionHeading";
import TextBox from "../../ui/elements/text-box/TextBox";
import Item from "../item/Item";

function HourlyWeather() {
  useFetchWeatherByLocation();
  const weatherData = useAppSelector(selectWeatherData);

  let hourlyForecast = <></>;
  if (weatherData.status === "loading") {
    hourlyForecast = <p>Loading...</p>;
  } else if (weatherData.status === "succeeded") {
    hourlyForecast = (
      <ul>
        {weatherData?.forecast?.forecastday[0].hour
          .filter(
            (hour) => hour.time_epoch >= weatherData.current.last_updated_epoch
          )
          .map((hour) => {
            const table = [
              { field: "RealFeel", value: hour.feelslike_c },
              { field: "Wind", value: hour.wind_kph },
              { field: "Wind Gust", value: hour.gust_kph },
              { field: "Humidity", value: hour.humidity },
              { field: "Dew Point", value: hour.dewpoint_c },
              { field: "UV", value: hour.uv },
            ];
            return (
              <Item
                key={hour.time_epoch}
                time={hour.time.split(" ")[1]}
                icon={hour.condition.icon}
                temp={hour.temp_c}
                humidity={hour.humidity}
                chanceOfRain={hour.chance_of_rain}
                extraTable={table}
              />
            );
          })}
      </ul>
    );
  } else if (weatherData.status === "failed" || weatherData.status === "idle") {
    hourlyForecast = <TextBox content={<>No data to display</>} />;
  }

  return (
    <>
      <SectionHeading
        content={<>Hourly Weather Forecast For {weatherData.location.name}</>}
      />
      {hourlyForecast}
    </>
  );
}

export default HourlyWeather;
