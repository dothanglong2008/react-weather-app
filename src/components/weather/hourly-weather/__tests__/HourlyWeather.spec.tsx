import renderer from "react-test-renderer";
import HourlyWeather from "../HourlyWeather";
import { Provider } from "react-redux";
import { store } from "../../../../app/store";

describe("Test HourlyWeather component", () => {
  it("renders HourlyWeather correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HourlyWeather />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
