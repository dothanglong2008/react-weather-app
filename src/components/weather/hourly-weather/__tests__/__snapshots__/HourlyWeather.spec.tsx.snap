// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HourlyWeather component renders HourlyWeather correctly 1`] = `
Array [
  <h2
    className="sectionHeading"
  >
    Hourly Weather Forecast For 
  </h2>,
  <p>
    No data to display
  </p>,
]
`;
